source/cppexpose/include/cppexpose/plugin/ComponentManager.h

```diff
@@ -225,12 +237,12 @@ class CPPEXPOSE_API ComponentManager
 
 
 protected:
-    std::vector<std::string>                   m_paths;               ///< Plugin paths (all)
-    std::vector<std::string>                   m_pathsInternal;       ///< Plugin paths (internal)
-    std::vector<std::string>                   m_pathsUser;           ///< Plugin paths (user defined)
-    std::vector<AbstractComponent *>           m_components;          ///< Available components
-    std::map<std::string, PluginLibrary *>     m_librariesByFilePath; ///< Available components by path
-    std::map<std::string, AbstractComponent *> m_componentsByName;    ///< Available components by name
+    std::vector<std::string>                              m_paths;               ///< Plugin paths (all)
+    std::vector<std::string>                              m_pathsInternal;       ///< Plugin paths (internal)
+    std::vector<std::string>                              m_pathsUser;           ///< Plugin paths (user defined)
+    std::vector<AbstractComponent *>                      m_components;          ///< Available components
+    std::map<std::string, std::unique_ptr<PluginLibrary>> m_librariesByFilePath; ///< Available components by path
+    std::map<std::string, AbstractComponent *>            m_componentsByName;    ///< Available components by name
```

Comment:
They are statically initialized once per component class via the `COMPONENT` macros, so I'd say their respective classes.